lib/rspec/support/object_inspector.rb

```diff
@@ -13,6 +13,8 @@ def self.inspect(object)
           "#{object.to_s 'F'} (#{object.inspect})"
         elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
           object.description
+        elsif object.class == RSpec::Mocks::Double
+          "Double (#{object.instance_variable_get(:@name)})"
```

Comment:
Yeah, it's messy.  Here's an alternate idea, that addresses both this issue and the messiness of `defined?(RSpec::Mocks::Double)`: add an API like `ObjectInspector.register(RSpec::Mocks::Double) { ... }` so that rspec-mocks can register the inspect output for `RSpec::Mocks::Double`.  That way, rspec-support doesn't need to know anything about rspec-mocks -- rspec-mocks would simply register itself so that its objects are formatted well when failure messages use the `ObjectInspector`.

Thoughts?